Sources/ScipioKit/Producer/FrameworkProducer.swift

```diff
@@ -136,6 +144,16 @@ struct FrameworkProducer {
         }
     }
 
+    private func makeBuildOptions(for product: BuildProduct, basedOn baseBuildOptions: BuildOptions, platformMatrix: PlatformMatrix) -> BuildOptions {
+        guard let overriddenSDKs = platformMatrix[product.target.name] else {
+            return baseBuildOptions
+        }
+
+        var newBuildOptions = baseBuildOptions
+        newBuildOptions.sdks = overriddenSDKs
+        return newBuildOptions
+    }
```

Comment:
It might be possible to add an extension of `BuildOptions`.

```swift
extension BuildOptions {
    fileprivate func overridingSDKs(for product: BuildProduct, platformMatrix: PlatformMatrix) -> BuildOptions {
        guard let overriddenSDKs = platformMatrix[product.target.name] else {
            return self
        }
        var newBuildOptions = self
        newBuildOptions.sdks = overriddenSDKs
        return newBuildOptions
    }
}
```
Then, we can represent our intention.

```swift
for product in libraryTargets {
    assert(product.target.type == .library)

    let buildOptionsForProduct = buildOptions.overridingSDKs(for: product, platformMatrix: platformMatrix)
    let compiler = Compiler(rootPackage: rootPackage, buildOptions: buildOptionsForProduct)

    let cacheSystem = CacheSystem(rootPackage: rootPackage,
                                      buildOptions: buildOptionsForProduct,
                                      outputDirectory: outputDir,
                                      storage: cacheStorage
    )
    // ...
}
```